detectable/src/main/java/com/synopsys/integration/detectable/detectables/bitbake/BitbakeExtractor.java

```diff
@@ -73,34 +72,27 @@ public Extraction extract(
         ExecutableTarget bash
     ) {
         List<CodeLocation> codeLocations = new ArrayList<>();
-
         BitbakeSession bitbakeSession = new BitbakeSession(executableRunner, bitbakeRecipesParser, sourceDirectory, buildEnvScript, sourceArguments, bash, toolVersionLogger, buildFileFinder, bitbakeEnvironmentParser);
         bitbakeSession.logBitbakeVersion();
         File buildDir = bitbakeSession.determineBuildDir();
         BitbakeEnvironment bitbakeEnvironment = bitbakeSession.executeBitbakeForEnvironment();
-        for (String packageName : packageNames) {
-            Map<String, String> imageRecipes = null;
+        ShowRecipesResults showRecipesResults;
+        try {
+            showRecipesResults = bitbakeSession.executeBitbakeForRecipeLayerCatalog();
+        } catch (IOException | ExecutableFailedException e) {
+            String msg = String.format("Error collecting recipe layer information from show-recipes command: %s", e.getMessage());
```

Comment:
You could just bubble these up - if this message helpful for debugging though, looks good.

Reply:
Done